semgrep/sol-rules.yaml

```diff
@@ -152,11 +153,28 @@ rules:
     severity: ERROR
     message: Use abi.encodeCall instead of abi.encodeWithSelector
     patterns:
-      - pattern: |
-          abi.encodeWithSelector(...);
+      - pattern-either:
+          - pattern: |
+              abi.encodeWithSelector(...);
+          - pattern: |
+              abi.encodeWithSignature(...);
     paths:
       exclude:
-        - packages/contracts-bedrock/test
+        - op-chain-ops/script/testdata/scripts/ScriptExample.s.sol
+        - packages/contracts-bedrock/test/L2/GasPriceOracle.t.sol
```

Comment:
Let's use the inline `// nosemgrep: <ruleid>` syntax for ignoring things as much as possible. I think all of these could be ignored that way instead of adding them to these excludes.